addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java

```diff
@@ -62,12 +65,52 @@
 
     private List<DeviceMessageListener> deviceStatusListeners = new CopyOnWriteArrayList<>();
 
-    private static byte seqNbr = 0;
-    private static RFXComTransmitterMessage responseMessage = null;
-    private Object notifierObject = new Object();
     private RFXComBridgeConfiguration configuration = null;
     private ScheduledFuture<?> connectorTask;
 
+
+    private class TransmitQueue {
+        private Queue<RFXComBaseMessage> queue = new LinkedBlockingQueue<RFXComBaseMessage>();
+
+
+        public synchronized void enqueue(RFXComBaseMessage msg) throws IOException {
+            boolean wasEmpty = queue.isEmpty();
+            if (queue.offer(msg)) {
+                if (wasEmpty) {
+                    send();
+                }
+            } else {
+                logger.error("Transmit queue overflow. Lost message: {}", msg);
```

Comment:
But nothing is offline. What's happened there is that the queue is full. It will drain. Future messages may succeed regardless of what thing they are for. There is no on-going problem that can be addressed here. All that's happened is that we couldn't handle that one specific message. Nothing more.